app.py

```diff
@@ -0,0 +1,332 @@
+
+
+def create_app(debug=True):
+    app = Flask(__name__)
+    '''We can store configuration elements in the app.config'''
+    app.config.update(get_config(os.environ['UICONFIG_SETTINGS']))
+    app.config["SUPPORTED_TYPES"] = ["json", "yaml", "toml"]
+    app.config["MAIN_CONFIG"] = {"version": 1, "enabled": True}
+
+    with app.app_context():
+        app.config["SCAN_CACHE"] = scan_olives(current_app.config["data"]["local_olive_dir"],
+                                               current_app.config["data"]["instances"],
+                                               current_app.config["data"].get("blacklist", []))
+        return app
+
+
+""" For state: Load configuration file assay_info.jsonconfig using setting of the current app"""
+def load_config() -> dict:
+    assay_config = {}
+    config_path = get_config(os.environ['UICONFIG_SETTINGS'])["data"]["assay_config_file"]
+    with open(os.path.join(os.path.dirname(sys.argv[0]), config_path), 'r') as f:
+        assay_config = json.load(f)
+    return assay_config
+
+
+""" For state: Load presets TODO: update them here if config changed"""
+def load_presets() -> dict:
+    preset_config = {}
+    preset_path = get_config(os.environ['UICONFIG_SETTINGS'])["data"]["preset_path"]
+    with open(os.path.join(os.path.dirname(sys.argv[0]), preset_path), 'r') as f:
+        preset_config = json.load(f)
+    return preset_config
+
+
+my_app = create_app()
+state.config = load_config()
+state.preset_list = load_presets()
+
+"""Pretty print json with arrays in single line"""
+def pretty_json(json_text: str) -> str:
+    jstring = re.sub(r'(\[)\n', r'\1', json_text)
+    jstring = re.sub(r'(\d\")\s+', r'\1', jstring)
+    jstring = re.sub(r'(\d\",)\s+', r'\1', jstring)
+    jstring = re.sub(r'(\[)\s+', r'\1', jstring)
+    jstring = re.sub(r'(\d\",)\n', r'\1', jstring)
+    return jstring
+
+"""ensure keys exist when making items in a nested dict"""
+def ensure(d, *keys):
+    for k in keys:
+        d = d.setdefault(k, {})
+    return d
+
+"""Check what we have enabled for a given project and return as a dict"""
+def obtain_enabled(project_json: dict) -> list:
+    boxes = []
+    for entry in project_json.keys():
+        if isinstance(project_json[entry], list):
+            boxes.append(dict(id=entry, status=True))
+    return boxes
+
+
+"""
+   this will take a staging version of config, sort and prettify it for printing
+"""
+def save_config(conf_data: dict, output_file: str):
+    try:
+        vetted_od = SessionState.deepsort_dict(conf_data)
+        with open(output_file, "w") as wfj:
+            jstring = json.dumps(vetted_od, indent=2, ensure_ascii=False)
+            jstring = re.sub(r'(\[)\n', r'\1', jstring)
+            jstring = re.sub(r'(\d\")\s+', r'\1', jstring)
+            jstring = re.sub(r'(\.\d\",)\s+', r'\1', jstring)
+            jstring = re.sub(r'(\[)\s+', r'\1', jstring)
+            jstring = re.sub(r'(\d\",)\n', r'\1', jstring)
```

Comment:
reimplementation of pretty_json()?

Reply:
It is for printing arrays in one line, I don't think there are any built-in functions which can do this

Reply:
I mean you have reimplemented pretty_json() from line 58 here rather than calling it.